src/utils/ItemPriceCalculator.js

```diff
@@ -0,0 +1,272 @@
+const PRICETYPE_UNIT = 1
+const PRICETYPE_UNIT_VOLUME = 2
+const PRICETYPE_UNIT_WEIGHT = 3
+const DIRTYFIELD_BASE_PRICE = 1
+const DIRTYFIELD_UNIT_PRICE = 2
+const DIRTYFIELD_QTY = 3
+const DIRTYFIELD_EXTENDED_PRICE = 4
+const DIRTYFIELD_UPO = 5
+const DIRTYFIELD_MDAMOUNT = 6
+const DIRTYFIELD_MDTYPE = 7
+const DIRTYFIELD_BASE_LABOR_PRICE = 8
+const DIRTYFIELD_TWPG = 9
+const DIRTYFIELD_TD = 10
+const MDTYPE_PCT = 1
+const MDTYPE_AMOUNT = 2
+
+class ItemPriceRow {
+  constructor(
+    _priceType,
+    _basePrice,
+    _volume,
+    _weight,
+    _unitPrice,
+    _upo,
+    _qty,
+    _extendedPrice,
+    _mdAmount,
+    _mdType,
+    _baseLaborPrice,
+    _totalWeightPerG,
+    _mdValue,
+    _tdPct,
+    _dirtyField
+  ) {
+    this.priceType = _priceType
+    this.basePrice = _basePrice
+    this.volume = _volume
+    this.weight = _weight
+    this.unitPrice = _unitPrice
+    this.upo = _upo
+    this.qty = _qty
+    this.extendedPrice = _extendedPrice
+    this.mdAmount = _mdAmount
+    this.mdType = _mdType
+    this.baseLaborPrice = _baseLaborPrice
+    this.totalWeightPerG = _totalWeightPerG
+    this.mdValue = _mdValue
+    this.tdPct = _tdPct
+    this.dirtyField = _dirtyField
+  }
+}
+
+const logIPR = itemPriceRow => {
+  console.log('priceType ' + itemPriceRow.priceType)
+  console.log('basePrice ' + itemPriceRow.basePrice)
+  console.log('baseLaborPrice ' + itemPriceRow.baseLaborPrice)
+  console.log('volume ' + itemPriceRow.volume)
+  console.log('unitPrice ' + itemPriceRow.unitPrice)
+  console.log('upo ' + itemPriceRow.upo)
+  console.log('qty ' + itemPriceRow.qty)
+  console.log('extendedPrice ' + itemPriceRow.extendedPrice)
+  console.log('mdAmount ' + itemPriceRow.mdAmount)
+  console.log('mdType ' + itemPriceRow.mdType)
+  console.log('mdValue ' + itemPriceRow.mdValue)
+  console.log('dirtyField ' + itemPriceRow.dirtyField)
```

Comment:
console log

Reply:
since they are in log function we need them